pkg/loki/modules.go

```diff
@@ -393,6 +393,10 @@ func (t *Loki) initQuerier() (services.Service, error) {
 		return nil, err
 	}
 
+  if t.Cfg.Pattern.MetricAggregation.Enabled {
```

Comment:
This works if running Loki all-in-one or sharing a config file, but I don't think it will work if the pattern-ingester and querier are running as separate components and this cfg is set using a flag on the pattern-ingester so the querier doesn't see it

Reply:
True, but since we already follow this pattern in many other places I think we've already created a dependency on a shared config file.

* [We check bloom gateway config in the index gateway](https://github.com/grafana/loki/blob/11c290e46a5f43f2e228816ee283524ded5588e5/pkg/loki/modules.go#L1519)
* [We already check pattern config in the querier](https://github.com/grafana/loki/blob/11c290e46a5f43f2e228816ee283524ded5588e5/pkg/loki/modules.go#L418)
* [We check ingester config in the querier](https://github.com/grafana/loki/blob/11c290e46a5f43f2e228816ee283524ded5588e5/pkg/loki/modules.go#L387)

 Would your preference be we specify `-querier.metric_aggregation_enabled` as a command line flag as well (as opposed to hiding it)?